example-crates/wallet_rpc/src/main.rs

```diff
@@ -0,0 +1,91 @@
+use bdk::{
+    bitcoin::{Address, Network},
+    wallet::{AddressIndex, Wallet},
+    SignOptions,
+};
+use bdk_bitcoind_rpc::{
+    bitcoincore_rpc::{Auth, Client, RpcApi},
+    Emitter,
+};
+use bdk_file_store::Store;
+use std::str::FromStr;
+
+const DB_MAGIC: &str = "bdk-rpc-wallet-example";
+const SEND_AMOUNT: u64 = 5000;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = std::env::args().collect::<Vec<_>>();
+    let db_path = std::env::temp_dir().join("bdk-rpc-example");
+    let db = Store::<bdk::wallet::ChangeSet>::new_from_path(DB_MAGIC.as_bytes(), db_path)?;
+
+    let external_descriptor = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/0/*)";
+    let internal_descriptor = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/1/*)";
+
+    let mut wallet = Wallet::new(
+        external_descriptor,
+        Some(internal_descriptor),
+        db,
+        Network::Testnet,
+    )?;
+
+    let address = wallet.get_address(AddressIndex::New);
+    println!("Generated Address: {}", address);
+
+    let balance = wallet.get_balance();
+    println!("Wallet balance before syncing: {} sats", balance.total());
+
+    let rpc_client = Client::new(&args[0], Auth::UserPass(args[1].clone(), args[2].clone()))?;
+
+    println!(
+        "Connected to Bitcoin Core RPC at {:?}",
+        rpc_client.get_blockchain_info().unwrap()
+    );
+
+    wallet.set_lookahead_for_all(args[3].parse::<u32>()?);
+
+    let chain_tip = wallet.latest_checkpoint();
+    let mut emitter = match chain_tip {
+        Some(cp) => Emitter::from_checkpoint(&rpc_client, cp),
+        None => Emitter::from_height(&rpc_client, args[4].parse::<u32>()?),
+    };
+
+    while let Some((height, block)) = emitter.next_block()? {
+        println!("Applying block {} at height {}", block.block_hash(), height);
+        wallet.apply_block_relevant(block, height)?;
+        wallet.commit()?;
+    }
+
+    let unconfirmed_txs = emitter.mempool()?;
+    println!("Applying unconfirmed transactions: ...");
+    wallet.batch_insert_relevant_unconfirmed(unconfirmed_txs.iter().map(|(tx, time)| (tx, *time)));
+    wallet.commit()?;
+
+    let balance = wallet.get_balance();
+    println!("Wallet balance after syncing: {} sats", balance.total());
+
+    if balance.total() < SEND_AMOUNT {
+        println!(
+            "Please send at least {} sats to the receiving address",
+            SEND_AMOUNT
+        );
+        std::process::exit(0);
```

Comment:
nit:
```suggestion
        std::process::exit(1);
```
otherwise, this is equivalent to `return Ok(())` afaik

Reply:
I used zero here instead of 1 because it is what is used in other examples. I have changed it to 1 though.